### DISCUSSION TITLE
2FOC V3.6 firmware Released

### DISCUSSION THREAD

**Discussion by ale-git** [robotology/community#657] [Releases]
Hi, the new 2FOC firmware is available [here](https://github.com/robotology/icub-firmware-build/blob/devel/CAN/2foc/2foc.experimentalV3.6.hex).

Features are:

- improved current sensing with self calibration at startup
- improved heuristics on fault detection
- temperature sensor support (yarprobotinterface support coming soon)
- motor encoder >16 bit resolution support
- smoother phase transition in HALL+QE robots
- solves the problem of autocalibration messages masked by the protocol
- back compatible with V3.3

**Notice**: it requires only one change in the `motorControl/*service.xml` files: now the motor encoder resolution must be the _counts per revolution_ provided by the encoder sensor before the 2FOC 4x interpolation. For example, the 14400 value written in the 3.3 configuration files becomes 3600 in the 3.6 configuration files.

**Comment by pattacini** [comment]
cc @robotology/everyone
